src/tests/ftest/soak/soak.py

```diff
@@ -65,239 +78,273 @@ def job_done(self, args):
-                    cmd = "srun -l --mpi=pmi2 --export={} {}".format(
-                        ",".join(exports), ior_cmd)
-                    command.append(cmd)
-                    self.log.debug("<<IOR cmdline >>: %s \n", cmd)
-        return command
-
-    def create_dmg_cmdline(self, job_params, job_spec, pool):
-        """Create a dmg cmdline to run in slurm batch.
+                    for o_type in oclass_list:
+                        ior_cmd = IorCommand()
+                        ior_cmd.namespace = ior_params
+                        ior_cmd.get_params(self)
+                        if iteration is not None and iteration < 0:
+                            ior_cmd.repetitions.update(1000000)
+                        if self.job_timeout is not None:
+                            ior_cmd.max_duration.update(self.job_timeout)
+                        else:
+                            ior_cmd.max_duration.update(10)
+                        ior_cmd.api.update(api)
+                        ior_cmd.block_size.update(b_size)
+                        ior_cmd.transfer_size.update(t_size)
+                        ior_cmd.daos_oclass.update(o_type)
+                        ior_cmd.set_daos_params(self.server_group, pool)
+                        # srun cmdline
+                        nprocs = self.nodesperjob * ppn
+                        env = ior_cmd.get_default_env("srun", self.tmp)
+                        if ior_cmd.api.value == "MPIIO":
+                            env["DAOS_CONT"] = ior_cmd.daos_cont.value
+                        cmd = Srun(ior_cmd)
+                        cmd.setup_command(env, None, nprocs)
+                        cmd.ntasks_per_node.update(ppn)
+                        commands.append(cmd.__str__())
+
+                        self.log.info(
+                            "<<IOR cmdline>>: %s \n", commands[-1].__str__())
+        return commands
+
+    def create_dfuse_cont(self, pool):
+        """Create a TestContainer object to be used to create container.
 
         Args:
-            job_params (str): job params from yaml file
-            job_spec (str): specific dmg job to run
+
+            pool (obj):   TestPool obj
+
         Returns:
-            cmd: [description]
+            cuuid: container uuid
 
         """
-        cmd = ""
-        return cmd
+        # TO-DO: use daos tool when available
+        # This method assumes that doas agent is running on test node
+        
+        cmd = "daos cont create --pool={} --svc={} --type=POSIX".format(
+            pool.uuid, ":".join(
+                [str(item) for item in pool.svc_ranks]))
+        try:
+            result = process.run(cmd, shell=True, timeout=30)
+        except process.CmdError as error:
+            raise SoakTestError(
+                "<<FAILED: Dfuse container failed {}>>".format(error))
+        self.log.info(
+            "Dfuse Container UUID = {}".format(result.stdout.split()[3]))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments